src/eval.rs

```diff
@@ -383,6 +383,34 @@ impl Expr {
 
 /// Used for merging sets during parsing. For example:
 /// { a.b = 1; a.c = 2; } => { a = { b = 1; c = 2; }; }
+///
+/// Nix only allows merging several such KeyValuePairs when they correspond
+/// to bare literals. Inserting a mere indirection through let or a function trips
+/// it of:
```

Comment:
Did you mean, "trips it up"? Maybe it would be more clear to say "Inserting a mere indirection through let or a function prevents this merging from happening"?

Reply:
done